extensions/vscode/src/entrypointTracker.ts

```diff
@@ -0,0 +1,214 @@
+// Copyright (C) 2023 by Posit Software, PBC.
+
+import {
+  Disposable,
+  TextDocument,
+  TextEditor,
+  commands,
+  window,
+  workspace,
+} from "vscode";
+import { Utils as uriUtils } from "vscode-uri";
+
+import { useApi } from "src/api";
+import { Contexts } from "src/constants";
+import { getPythonInterpreterPath } from "src/utils/config";
+import { isActiveDocument, relativeDir } from "src/utils/files";
+import { hasKnownContentType } from "src/utils/inspect";
+import { getSummaryStringFromError } from "src/utils/errors";
+
+/**
+ * Determines if a text document is an entrypoint file.
+ *
+ * @param document The text document to inspect
+ * @returns If the text document is an entrypoint
+ */
+async function isDocumentEntrypoint(document: TextDocument): Promise<boolean> {
+  const dir = relativeDir(document.uri);
```

Comment:
This was quite tricky digging into what VSCode could do for us and what else is out there. I broke it out into its own utils function so we can get a `dir` path for any API requests.